extensions/amp-analytics/0.1/cid-impl.js

```diff
@@ -237,45 +237,35 @@ function getBaseCid(cid, persistenceConsent) {
     return cid.baseCid_;
   }
   const win = cid.win;
-  const stored = read(win);
-  // See if we have a stored base cid and whether it is still valid
-  // in terms of expiration.
-  if (stored && !isExpired(stored)) {
-    if (shouldUpdateStoredTime(stored)) {
-      // Once per interval we mark the cid as used.
-      store(win, stored.cid);
-    }
-    return cid.baseCid_ = Promise.resolve(stored.cid);
-  }
-  // If we are being embedded, try to get the base cid from the viewer.
-  // Note, that we never try to persist to localStorage in this case.
-  const viewer = viewerFor(win);
-  if (viewer.isIframed()) {
-    return cid.baseCid_ = viewer.getBaseCid().then(cidValue => {
-      if (!cidValue) {
-        throw new Error('No CID');
+
+  return read(win).then(stored => {
+    let needsToStore = false;
+
+    // See if we have a stored base cid and whether it is still valid
+    // in terms of expiration.
+    if (stored && !isExpired(stored)) {
+      cid.baseCid_ = Promise.resolve(stored.cid);
+      if (shouldUpdateStoredTime(stored)) {
+        needsToStore = true;
       }
-      return cidValue;
-    });
-  }
+    } else {
+      // We need to make a new one.
+      cid.baseCid_ = cryptoFor(win)
+          .then(crypto => crypto.sha384Base64(getEntropy(win)));
+      needsToStore = true;
+    }
 
-  // We need to make a new one.
-  cid.baseCid_ = cryptoFor(win)
-      .then(crypto => crypto.sha384Base64(getEntropy(win)));
+    if (needsToStore) {
+      // Storing the value may require consent. We wait for the respective
+      // promise.
+      Promise.all([cid.baseCid_, persistenceConsent])
```

Comment:
Currently viewer doesn't ask for consent before persisting an ID. I can change the logic here to match the behavior, but was it intentional?